### PR TITLE
(DEBUG BRANCH) Approve shall be applied on ER20Vault not translator

### PR DESCRIPTION
**Pull request by adaki2004**
@Brechtpd 
This is where i'm at.

I use the command for testing. For whatever reason the 'minting' is not working.  (You can clearly see my debug prints :))
`forge test --match-test test_20Vault_bridge_native_usdc_back_to_l1 -vv`

Any idea what am i doing wrong with the delegateCall.. ?

### CODE REVIEW

File: packages/protocol/contracts/tokenvault/erc20/translators/UsdcTranslator.sol [hunk 1]
```diff
@@ -27,32 +31,46 @@ contract UsdcTranslator is BaseTranslator {
...
     /// @param amount The amount of tokens to mint.
-    function mint(address account, uint256 amount) public onlyFromNamed("erc20_vault") {
-        IERC20TokenVaultRelayer(nativeTokenContract).mint(account, amount);
+    function mint(
+        address token,
+        address account,
+        uint256 amount
+    )
+        public
+        onlyFromNamed("erc20_vault")
```
Comment (Brechtpd) [inline packages/protocol/contracts/tokenvault/erc20/translators/UsdcTranslator.sol:40]: I think you can just remove this check, the important things is that msg.sender will be the ERC20Vault for the USDC token contract, so only delegate calls done by ERC20Vault will have that property.

**Comment by adaki2004** [comment]
https://ethereum.stackexchange.com/questions/125217/delegatecall-failing -> Tried, not helped.

**Comment by adaki2004** [comment]
Storage layout:
kép

**Comment by adaki2004** [comment]
Storage layout now - and still, `delegatecall()` does not resolve the mint function properly.
kép

**Pull request merged**